core/blockchain/blockchain_gateway.py

```diff
@@ -0,0 +1,355 @@
+                "http://localhost:{0}/txs".format(self.port), json=tx)
+            tx_receipt.raise_for_status()
+        except Exception as e:
+            logging.info("HTTP POST error, got: {0}".format(e))
+        return tx_receipt.text
+
+    def getter(self, key: str) -> list:
+        """
+        Next, provided a key, get the IPFS hash
+        from the blockchain and download the object from IPFS
+        """
+        logging.info("Getting from blockchain...")
+        self.update_state(self.get_global_state())
+        retval = self._download(key)
+        return retval
+
+    def _upload(self, obj: object) -> str:
+        """
+        Provided any Python object, store it on IPFS and then upload
+        the hash that will be uploaded to the blockchain as a value
+        """
+        ipfs_hash = self._content_to_ipfs(obj)
+        return str(ipfs_hash)
+
+    def _download(self, key: str) -> object:
+        """
+        Provided an on-chain key, retrieve the value from local state and
+        retrieve the Python object from IPFS
+        TODO: implement a better way to parse through state list
+        """
+        relevant_txs = [self._ipfs_to_content(tx.get('content'))
+                            for tx in self.state if (tx.get('key') == key)]
+        return relevant_txs
+
+    def _ipfs_to_content(self, ipfs_hash: str) -> object:
+        """
+        Helper function to retrieve a Python object from an IPFS hash
+        """
+        return self.client.get_json(ipfs_hash)
+
+    def _content_to_ipfs(self, content: dict) -> str:
+        """
+        Helper function to deploy a Python object onto IPFS, 
+        returns an IPFS hash
+        """
+        ipfs_hash = self.client.add_json(content)
+        return ipfs_hash
+
+    ##########################################################################
+    ###                         DEVELOPER SECTION                          ###
+    ##########################################################################
+
+    def broadcast_decentralized_learning(self, model_config: object)-> str:
+        """
+        Upload a model config and weights to the blockchain
+        """
+        tx_receipt = self.setter(None, model_config, True)
+        return tx_receipt
+
+    def broadcast_terminate(self, key: str) -> str:
+        """
+        Terminates decentralized training
+        TODO: check if training even started
+        """
+        tx_receipt = self.setter(key, None)
+        return tx_receipt
+
+    def handle_decentralized_learning_owner(self, model_config: object) -> None:
+        """
+        Return weights after training terminates
+        TODO: add condition to check if training for specific model terminated
+        """
+        final_weights = self.getter(header)
+        return final_weights
+
+    ##########################################################################
+    ###                          PROVIDER SECTION                          ###
```

Comment:
Everything below this section (and above if I missed it) should use the same new python syntax as the sections above. Let's make the class consistent in style.

Reply:
Also, please add docstrings to all methods.